chess/events/src/index.ts

```diff
@@ -0,0 +1,84 @@
+import type { EventQueue } from '@paima/events';
+import { generateAppEvents } from '@paima/events';
+import { genEvent } from '@paima/events';
+import { Type } from '@sinclair/typebox';
+
+export const CreatedLobby_v1 = genEvent({
+  name: 'CreatedLobby',
+  fields: [
+    {
+      indexed: true,
+      name: 'user',
+      type: Type.String(),
+    },
+    {
+      indexed: true,
+      name: 'lobbyId',
+      type: Type.String(),
+    },
+    {
+      indexed: false,
+      name: 'rounds',
+      type: Type.Number(),
+    },
+  ],
+} as const);
+
+export const CreatedLobby_v2 = genEvent({
+  name: 'CreatedLobby',
+  fields: [
+    {
+      indexed: true,
+      name: 'user',
+      type: Type.String(),
+    },
+    {
+      indexed: true,
+      name: 'lobbyId',
+      type: Type.String(),
+    },
+    {
+      indexed: false,
+      name: 'rounds',
+      type: Type.Number(),
+    },
+    {
+      indexed: false,
+      name: 'isPractice',
+      type: Type.Boolean(),
+    },
```

Comment:
It looks like v1 & v2 where defined to have this optional type `Type.Optional(Type.Boolean())` 
It might be better for the insistence as both events have the same name and indexed fields (and there is no way of distinguishing them)

Reply:
They can be distinguished by the topic (which is at `app/topic`). Since the topic is the hash of the signature, these will have different hashes, because one has one extra field.

But you could define a single event with the optional field as you say. The thing is that events are immutable, you can't add a field to an existing one, so this was an example of what you would need to do if you want to add a new field. So let's say you forgot to add the field initially, you can add this later for example.

Or if you want to change the type of an existing field you would need to do something similar.